src/utils/private/buffer_impl.cpp

```diff
@@ -364,6 +364,12 @@ class Buffer::BufferImpl {
 
 
 	const short* getSamplesRawInterleaved(int nb_samples)
+	{
+		const short* p_dat = (const short*) getSamplesRawInterleavedVoid(nb_samples);
```

Comment:
why not just do `return getSamplesRawInterleavedVoid(nb_samples);` ?

at least C [I think C++ as well] doesn't complain if you allow implicit casts to/from `void *`
another mechanism, may be to use `reinterpret_cast`; i am not sure about it's added value of C-type casts, but it looks more C++-ish

Reply:
C++ allows implicit conversions TO a void * type, but to convert FROM a void * type requires an explicit cast. In this case (const short *) if you're using c style cast, or an static_cast if you're using c++ style casts.